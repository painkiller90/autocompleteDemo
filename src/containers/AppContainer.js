import { useState, useEffect } from 'react';
import debounce from 'debounce';
import { InputSearch } from '../components/InputSearch';
import { SearchResults } from '../components/SearchResults';
import './appContainer.css';

const AppContainer = () => {
  const [queryString, setQueryString] = useState('');
  const [issuesData, setIssuesData] = useState([]);
  const [filteredData, setFilteredData] = useState([]);

  const fetchGithubData = () => {
    fetch('https://api.github.com/repos/facebook/react/issues')
      .then((data) => data.json())
      .then((json) => {
        setIssuesData(json);
      });
  };

  const filterResults = (queryString) => {
    if (Array.isArray(issuesData) && queryString.length > 1) {
      const regex = new RegExp(queryString, 'gi');
      return issuesData.filter((obj) => obj.title.match(regex));
    }
    return [];
  };

  const debounceSearch = debounce((inputValue) => {
    const filteredResults = filterResults(inputValue);
    setFilteredData(filteredResults);
    setQueryString(inputValue);
  }, 500);

  const handleSearch = ({ target }) => {
    const { value } = target;
    debounceSearch(value);
  };

  useEffect(() => {
    fetchGithubData();
  }, []);

  return (
    <div className='app-container'>
      <form id='search' role='search'>
        <InputSearch
          placeholder='Search'
          handleSearch={handleSearch}
          ariaControlId={'search-results-id'}
        />
        <SearchResults
          results={filteredData}
          queryString={queryString}
          ariaControlId={'search-results-id'}
        />
      </form>
    </div>
  );
};

export default AppContainer;
