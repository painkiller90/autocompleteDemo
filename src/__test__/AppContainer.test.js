import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import AppContainer from '../containers/AppContainer';

describe('App Container tests', () => {
  const renderComponent = () => {
    return render(
      <div role='main'>
        <AppContainer />
      </div>
    );
  };

  it('should render AppContainer properly without any prop', () => {
    const Container = render(<AppContainer />).baseElement;
    expect(Container).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const Container = renderComponent().baseElement;
    expect(await axe(Container)).toHaveNoViolations();
  });
});
