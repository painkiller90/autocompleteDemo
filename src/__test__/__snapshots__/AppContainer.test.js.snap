// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container tests should render AppContainer properly without any props 1`] = `
<body>
  <div>
    <div
      class="app-container"
    >
      <form
        id="search"
        role="search"
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="search-results-id"
          id="input-search"
          placeholder="Search"
          type="search"
        />
        <ul
          class="search-result"
          id="search-results-id"
        />
      </form>
    </div>
  </div>
</body>
`;

exports[`App Container tests should render Tree container properly without any childrens and props 1`] = `
<body>
  <div>
    <div
      class="app-container"
    >
      <form
        id="search"
        role="search"
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="search-results-id"
          id="input-search"
          placeholder="Search"
          type="search"
        />
        <ul
          class="search-result"
          id="search-results-id"
        />
      </form>
    </div>
  </div>
</body>
`;
